lib/ansible/plugins/connection/docker.py

```diff
@@ -132,8 +132,8 @@ def _get_docker_version(self):
 
         cmd, cmd_output, err, returncode = self._old_docker_version()
         if returncode == 0:
-            for line in cmd_output.split('\n'):
-                if line.startswith('Server version:'):  # old docker versions
+            for line in cmd_output.split(b'\n'):
+                if line.startswith(b'Server version:'):  # old docker versions
```

Comment:
In my opinion it's easier to just stay with the current type. Since Popen natively returns bytes, it doesn't seem like a worthwhile thing to convert it to `to_text` or `to_native` just so we don't have to use `b''` somewhere.  I feel like the current implementation is less invasive.

We also don't need to do any conversions after `_new_docker_version`.  `_sanitize_version` has been updated to use bytes also.

The only place where we couldn't use bytes is in the actual version comparisons, due to the regex implemented in distutils.